mantidproject/mantid
# Fix SANS bug when processing with no file types selected

**Description of work.**
ISIS SANS has three output modes: memory (posting workspaces to logger), file (save workspaces to disk), and both.

This PR raises a clear warning box when you process data in **file** or **both** modes with no file types selected. Furthermore, you can now successfully process with no file types selected while in **memory** mode.

Finally, we disable the file type buttons when in **memory** mode, to make it clear that this mode does not save to file

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. Load the attached user file and batch file
3. At the bottom of the gui, select the **file** radio button
4. Deselect all of **canSAS**, **NXcanSAS**, and **RKH** buttons
5. At the top of the gui, click `Process All`
6. A warning box should appear which clearly explains why the interface has not processed the data
7. Select the **both** radio button and process again. The same warning box should appear
8. Select the **memory** radio button. The file type buttons should be disabled.
9. Process again. Processing should be successful (previously, an uncaught exception was raised)

Fixes #21975 
Fixes #25637 

**Data:**
[loqPRData.zip](https://github.com/mantidproject/mantid/files/3181420/loqPRData.zip)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
